patternlab/styleguide/source/assets/js/modules/mainNavHamburger.js

```diff
@@ -1,124 +1,156 @@
 const body = document.querySelector("body");
 let width = body.clientWidth;
-const menuButton = document.querySelector(".ma__header__hamburger .js-header-menu-button");
-let menuButtonText = document.querySelector('.ma__header__hamburger .js-header__menu-text');
+const menuButton = document.querySelector(".js-header-menu-button");
+let menuButtonText = document.querySelectorAll(".js-header__menu-text");
 let buttonLabel;
-const feedbackButton = document.querySelector('.ma__fixed-feedback-button');
-const mainMenu = document.querySelector('.ma__header__hamburger .js-main-nav-hamburger');
-let menuItems = document.querySelectorAll('.ma__header__hamburger .js-main-nav-hamburger-toggle');
-const menuOverlay = document.querySelector('.menu-overlay');
-const searchForm = document.querySelector(".ma__header__hamburger .js-header-search-menu .js-header-search-form");
-let utilNavWide = document.querySelector('.ma__header__hamburger .js-utility-nav--wide');
-
+const hamburgerMenuContainer = document.querySelector(".ma__header__hamburger__nav-container");
+const feedbackButton = document.querySelector(".ma__fixed-feedback-button");
+let menuItems = document.querySelectorAll(".js-main-nav-hamburger-toggle");
+const menuOverlay = document.querySelector(".menu-overlay");
+let utilNavWide = document.querySelector(".js-utility-nav--wide");
+const jumpToSearchButton = document.querySelector(".js-header-search-access-button");
+const hamburgerSearchInput = document.getElementById("nav-search");
 
 if (null !== menuButtonText) {
   buttonLabel = menuButtonText.textContent;
 }
+
 // Open and close the menu
-if (null !== menuButton) {
+// if (null !== menuButton) {
   menuButton.addEventListener("click", function (event) {
     event.preventDefault();
 
-    if (body.classList.contains("show-menu-hamburger")) {
-      closeMenu();
-    } else {
-      menuButton.focus();
-      openMenu();
-    }
-  });
+    // This control the visibility of the dropdown to keyboard and screen reader users while maintaining the show/hide animation effect.
+    hamburgerMenuContainer.toggleAttribute("aria-hidden");
 
-  menuButton.addEventListener('keydown', function (e) {
+    if (body.classList.contains("show-menu")) {
+      closeMenu();
+      document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .goog-te-menu-value").removeAttribute("tabindex");
+      document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .direct-link").removeAttribute("tabindex");
+      document.querySelector(".js-utility-nav--wide .ma__utility-nav__item  .js-util-nav-toggle").removeAttribute("tabindex");
 
-    if (e.code == 'ArrowDown') {
-      event.preventDefault();
+      setTimeout(function timeoutFunction() {
```

Comment:
```suggestion
      // Setting a timeout function to allow for the closing of the menu to occur first.
      setTimeout(function timeoutFunction() {
```
I wonder if we should add a comment like the one above here to explain the need for the timeout.